### DISCUSSION TITLE
The Next.js example is very outdated (using next 9)

### DISCUSSION THREAD

**Discussion by everdrone** [algolia/instantsearch#5311] [Q&A] [answered]
I'm trying to integrate InstantSearch on a next 12 app but I'm spending more time trying to translate the [next v9 example](https://github.com/algolia/react-instantsearch/tree/master/examples/next) into next v12 than actually working on the search itself.

The class components have multiple functions that have been replaced with hooks and `getServerSideProps`. Plus the `widgetsCollector` parameter does not appear in the example.

Does anyone have a working codesandbox or repo where Next.js 12 works with url parameters and pagination?

**Comment by Haroenv** [comment, marked as answer]
I've updated the example in https://github.com/algolia/react-instantsearch/pull/3438, you can find it here before the PR is merged: https://github.com/algolia/react-instantsearch/tree/chore/update-next-example/examples/next

**Reply by everdrone** [reply]
Thank you! Just tested it and works like a charm!
